src/lib/custom-deploy/custom-deploy.ts

```diff
@@ -14,6 +16,32 @@ export function isSupportedApp( app: App ): boolean {
 	return WORDPRESS_SITE_TYPE_IDS.includes( app.typeId as number );
 }
 
+export async function validateCustomDeployKey(
+	customDeployKey: string,
+	envId: number
+): Promise< void > {
+	if ( customDeployKey.length === 0 ) {
+		exit.withError( 'Valid custom deploy key is required.' );
+	}
+
+	const VALIDATE_CUSTOM_DEPLOY_ACCESS_MUTATION = gql`
+	mutation ValidateCustomDeployAccess {
+		validateCustomDeployAccess( input: { environmentIds: ${ envId } } ) {
+			success
+		}
+	}
+`;
+
+	const api = await API( { customAuthToken: customDeployKey } );
+	try {
+		await api.mutate( { mutation: VALIDATE_CUSTOM_DEPLOY_ACCESS_MUTATION } );
```

Comment:
rather than modify `API`, it might be cleaner to just pass in the header here (and wherever else API might be called in the context of custom deploy). untested but i think this will work
```suggestion
		await api.mutate( {
			mutation: VALIDATE_CUSTOM_DEPLOY_ACCESS_MUTATION,
			context: {
				headers: {
					Authorization: `Bearer ${ customDeployKey }`,
				},
			},
		} );
```